### PR TITLE
Convert ReactDOMTestSelectors-test.js to createRoot

### PR DESCRIPTION
**Pull request by rickhanlonii**
Straightforward adding createRoot and act

### CODE REVIEW

File: packages/react-dom/src/__tests__/ReactDOMTestSelectors-test.js [hunk 1]
```diff
@@ -31,7 +31,8 @@ describe('ReactDOMTestSelectors', () => {
     jest.resetModules();
 
     React = require('react');
-    act = React.unstable_act;
+    createRoot = require('react-dom/client').createRoot;
+    act = require('internal-test-utils').act;
```
Comment (rickhanlonii) [inline packages/react-dom/src/__tests__/ReactDOMTestSelectors-test.js:35]: Switch to internal act

File: packages/react-dom/src/__tests__/ReactDOMTestSelectors-test.js [hunk 2]
```diff
@@ -1389,7 +1493,7 @@ No matching component was found for:
         {rect: rect1, ratio: 1},
       ]);
 
-      act(() => increment());
+      await act(() => increment());
```
Comment (rickhanlonii) [inline packages/react-dom/src/__tests__/ReactDOMTestSelectors-test.js:1496]: Need to await act now

File: packages/react-dom/src/__tests__/ReactDOMTestSelectors-test.js [hunk 3]
```diff
@@ -1412,7 +1516,7 @@ No matching component was found for:
         {rect: rect2, ratio: 0.25},
       ]);
 
-      act(() => increment());
+      await act(() => increment());
```
Comment (rickhanlonii) [inline packages/react-dom/src/__tests__/ReactDOMTestSelectors-test.js:1519]: Need to await act now

**Review by kassens — APPROVED** [review]
Thanks!

**Comment by kassens** [comment]
Actually looks like some test failures that need to be fixed before merging.

**Comment by rickhanlonii** [comment]
Damn I thought i could sneak the failures by you

File: packages/react-dom/unstable_testing.js [hunk 4]
```diff
@@ -7,7 +7,30 @@
...
+  prefetchDNS,
+  preconnect,
+  preload,
+  preloadModule,
+  preinit,
+  preinitModule,
+  version,
+} from './index.js';
+
+export {createRoot, hydrateRoot} from './client.js';
```
Comment (rickhanlonii) [inline packages/react-dom/unstable_testing.js:32]: Heads up @gnoff, I needed to update the imports in the unstable_testing entrypoints so using create/hydrateRoot doesn't error. Let me know if there are any concerns, but I think this is what we want to do, at least while this is unstable.

**Pull request merged**